R/hmde_assign_data.R

```diff
@@ -1,30 +1,89 @@
-#' Assign data to template
+#' Assign data to template for chosen model
 #'
 #' @param model_template output from hmde_model
-#' @param ... data-masking name-value pairs
+#' @param data Input data tibble with columns including time, y_obs, obs_index, and additionally ind_id for multi-individual models
+#' @param step_size Step size for numerical integration.
+#' @param ... data-masking name-value pairs allowing specific input of elements
 #'
 #' @return updated named list with your data assigned to Stan model parameters
 #' @export
 
-hmde_assign_data <- function(model_template, ...){
-  # Grab user expressions
-  user_code <- rlang::enquos(..., .check_assign = TRUE)
+hmde_assign_data <- function(model_template, data = NULL, step_size = NULL, ...){
+  if(!model_template$model %in% hmde_model_name()){
+    stop("Model name not recognised. Run hmde_model_name() to see available models.")
```

Comment:
nice work adding all these checks. I didn't look thoroughly, but do you have tests associated with this? Can be good to add a test that checks it fails as intended, e.g.

```
expect_error(hmde_assign_data("Not a model"))
```

Reply:
I went back and checked, had a couple I forgot to add.